rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -588,83 +602,144 @@ private Integer addFederatorPublicKeyMultikey(boolean dryRun, BtcECKey btcKey, E
+        Instant federationCreationTime = Instant.ofEpochMilli(rskExecutionBlock.getTimestamp());
+        long federationCreationBlockNumber = rskExecutionBlock.getNumber();
+
+        return pendingFederation.buildFederation(federationCreationTime, federationCreationBlockNumber, constants, activations);
+    }
+
+    private void clearPendingFederationVoting() {
+        // Clear pending federation and votes on election
+        provider.setPendingFederation(null);
         provider.getFederationElection(constants.getFederationChangeAuthorizer()).clear();
+    }
 
-        Federation oldFederation = provider.getOldFederation(constants, activations);
+    private void preserveFederationChangeInfo(Federation activeFederation) {
+        provider.setNextFederationCreationBlockHeight(rskExecutionBlock.getNumber());
 
-        if (activations.isActive(RSKIP186)) {
-            // Preserve federation change info
-            long nextFederationCreationBlockHeight = rskExecutionBlock.getNumber();
-            provider.setNextFederationCreationBlockHeight(nextFederationCreationBlockHeight);
-            Script oldFederationP2SHScript = activations.isActive(RSKIP377) && activeFederation instanceof ErpFederation ?
-                ((ErpFederation) activeFederation).getDefaultP2SHScript() : activeFederation.getP2SHScript();
-            provider.setLastRetiredFederationP2SHScript(oldFederationP2SHScript);
-        }
+        Script activeFederationMembersP2SHScript = getFederationMembersP2SHScript(activeFederation);
+        provider.setLastRetiredFederationP2SHScript(activeFederationMembersP2SHScript);
+    }
 
-        Federation newFederation = provider.getNewFederation(constants, activations);
+    private Script getFederationMembersP2SHScript(Federation federation) {
+        // when the federation is a standard multisig,
+        // the members p2sh script is the p2sh script
+        if (!activations.isActive(RSKIP377)) {
+            return federation.getP2SHScript();
+        }
+        if (!(federation instanceof ErpFederation)) {
+            return federation.getP2SHScript();
+        }
 
-        logger.debug("[commitFederation] New Federation committed: {}", newFederation.getAddress());
-        eventLogger.logCommitFederation(rskExecutionBlock, oldFederation, newFederation);
+        // when the federation also has erp keys,
+        // the members p2sh script is the default p2sh script
+        return ((ErpFederation) federation).getDefaultP2SHScript();
+    }
 
-        return FederationChangeResponseCode.SUCCESSFUL.getCode();
+    private void logCommitmentWithVotedFederation(BridgeEventLogger eventLogger, Federation federationToBeRetired, Federation votedFederation) {
+        eventLogger.logCommitFederation(rskExecutionBlock, federationToBeRetired, votedFederation);
+        logger.debug("[logCommitmentWithVotedFederation] Voted federation committed: {}", votedFederation.getAddress());
```

Comment:
Why debug log level?

Reply:
same, thats just how it was before. See [previous code](https://github.com/rsksmart/rskj/blob/feature/powpeg_validation_protocol/rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java#L664)